cylc/flow/scripts/cat_log.py

```diff
@@ -291,7 +333,7 @@ def get_option_parser() -> COP:
     parser.add_option(
         "-s", "--submit-number", "-t", "--try-number",
         help="Job submit number (default=%s, i.e. latest)." % NN,
-        metavar="INT", action="store", dest="submit_num", default=NN)
+        metavar="INT", action="store", dest="submit_num", default=None)
```

Comment:
Apply the default later on to allow specifying the job submission number in the ID.